installinstallmacos.py

```diff
@@ -30,42 +30,114 @@
 import plistlib
 import subprocess
 import sys
-import time
 import urlparse
 import xattr
+import re
 from xml.dom import minidom
 from xml.parsers.expat import ExpatError
+from distutils.version import LooseVersion
 
 
-DEFAULT_SUCATALOG = (
-    'https://swscan.apple.com/content/catalogs/others/'
-    'index-10.13seed-10.13-10.12-10.11-10.10-10.9'
-    '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog.gz')
+DEFAULT_SUCATALOGS = {
+    '17': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+    '18': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.14-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+}
+
 
 SEED_CATALOGS_PLIST = (
     '/System/Library/PrivateFrameworks/Seeding.framework/Versions/Current/'
     'Resources/SeedCatalogs.plist'
 )
 
 
+def get_board_id():
+    '''Gets the local system board ID'''
+    ioreg_cmd = ['/usr/sbin/ioreg', '-p', 'IODeviceTree', '-r', '-n', '/', '-d', '1']
+    try:
+        ioreg_output = subprocess.check_output(ioreg_cmd).splitlines()
+        for line in ioreg_output:
+            if 'board-id' in line:
+                board_id = line.split(" ")[-1]
+                board_id = board_id[board_id.find('<"')+2:board_id.find('">')]
+                return board_id
+    except subprocess.CalledProcessError, err:
+        print "Board ID could not be identified. Error was: %s" % err
+        return ''
```

Comment:
Still can fall through and return None if 'board-id' is never found.

Reply:
Could I just change this to a non-specific `except:`?

Reply:
That would not fix the issue I'm talking about, which is simply not finding "board-id" in the output. Perhaps this would never ever happen. But it's easy to make the code behave better in case it does.

Reply:
I think I resolved this now. None of the definitions that I've added would now return `None` if they failed to identify the value they are looking for. They now return `Not identified` so that the outputted text is informative.